### DISCUSSION TITLE
Is there a way to tell renovate to scan my separate branch for renovate.json instead master?

### DISCUSSION THREAD

**Discussion by gjorgi93** [renovatebot/renovate#9336] [Request Help]
I’m implementing renovate in one of our enterprise projects, i added the renovate job and renovate.json file in a separate branch, and my problem is that when the job is being executed, renovate scans the default master branch for possible renovate.json file instead of my separate branch, thus creates “onboarding” merge request which fails on our git hooks requirements. The goal is to test renovate on a specific branch before merging it to master and ommit the onboarding MR. 

This is how my renovate command in job template looks like:
`      renovate
      --platform="gitlab"
      --endpoint="${CI_API_V4_URL}"
      --git-author="${RENOVATE_GIT_AUTHOR}"
      --dry-run="${RENOVATE_DRY_RUN}"
      --labels="renovate"
      --token="${RENOVATE_TOKEN}"
      --enabled-managers="$ENABLED_MANAGERS"
      --prune-stale-branches="false"
      --onboarding="false"
      "${PROJECT_PATH}"`

**Comment by rarkins** [comment]
If you are to use per-repo config (default behavior, and recommended), then that config file (e.g. `renovate.json`) needs to be in the default branch, but can be configured to only scan alternative branch(es) by configuring `baseBranches`.

If all you care about is testing/dry-run, then it might work to add it to your command like so:

`      renovate
      --platform="gitlab"
      --endpoint="${CI_API_V4_URL}"
      --git-author="${RENOVATE_GIT_AUTHOR}"
      --dry-run="${RENOVATE_DRY_RUN}"
      --labels="renovate"
      --token="${RENOVATE_TOKEN}"
      --enabled-managers="$ENABLED_MANAGERS"
      --prune-stale-branches="false"
      --base-branches="next"
      --onboarding="false"
      "${PROJECT_PATH}"`


Note that there will be no branches or PRs added if dry-run is true, so you'd need to check the INFO logs to know what's planned.

**Reply by gjorgi93** [reply]
dry-run option is set to false, seems like renovate command does not support such option:
`error: unknown option '--base-branches=my-test-branch'`

**Reply by rarkins** [reply]
Sorry, my bad - it was intended as a `renovate.json` field and is disabled in CLI: https://github.com/renovatebot/renovate/blob/1e8c688a3516d2d579986df3bc71750bf17a12fb/lib/config/definitions.ts#L587

Seems we didn't disable for env so you could try ` RENOVATE_BASE_BRANCHES=my-test-branch` in env and that may work.

If your intentions are testing, can you simply fork it and test against the fork instead?
